clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -2023,38 +2077,44 @@ raw2trace_t::append_bb_entries(raw2trace_thread_data_t *tdata,
                 }
             }
         }
-        if (!skip_icache && record_encoding_emitted(tdata, decode_pc)) {
-            if (!append_encoding(tdata, decode_pc, instr->length(), buf, buf_start))
-                return false;
-        }
-
-        // XXX i#1729: make bundles via lazy accum until hit memref/end, if
-        // we don't need encodings.
-        buf->type = instr->type();
-        if (buf->type == TRACE_TYPE_INSTR_MAYBE_FETCH) {
-            // We want it to look like the original rep string, with just one instr
```

Comment:
It may be best to have a rep string fault test to ensure it rolls back to the right point.